usr/tgtd.c

```diff
@@ -817,6 +819,61 @@ static int lun_create(const _ha_request *reqp,
 	return HA_CALLBACK_CONTINUE;
 }
 
+static int new_stord(const _ha_request *reqp,
+	_ha_response *resp, void *userp)
+{
+	char *data = NULL;
+
+	char *hyc_argv[1]   = {"tgtd"};
+	uint16_t stord_port = 0;
+
+	data = ha_get_data(reqp);
+	if (data == NULL) {
+		set_err_msg(resp, TGT_ERR_NO_DATA,
+			"json config not given");
+		return HA_CALLBACK_CONTINUE;
+	}
+
+	json_error_t error;
+	json_auto_t *root = json_loads(data, 0, &error);
+
+	free(data);
+
+	if (root == NULL) {
+		set_err_msg(resp, TGT_ERR_INVALID_JSON,
+			"json config is incorrect");
+		return HA_CALLBACK_CONTINUE;
+	}
+
+	json_t *sip = json_object_get(root, "StordIp");
+	if (!json_is_string(sip)) {
+		set_err_msg(resp, TGT_ERR_INVALID_STORD_IP,
+			"StordIp is not string");
+		return HA_CALLBACK_CONTINUE;
+	}
+
+	json_t *sport = json_object_get(root, "StordPort");
+	if (!json_is_string(sport)) {
+		set_err_msg(resp, TGT_ERR_INVALID_STORD_PORT,
+			"StordPort is not string");
+		return HA_CALLBACK_CONTINUE;
+	}
+
+	int ret = str_to_int_range((char *)json_string_value(sport),
```

Comment:
Comments .. adhering to existing coding policies but the port range can go upto 64k.

Reply:
Done